### PR TITLE
Accessibility improvements: Button elements & a rel attributes

### LINKED ISSUE #45: Accessibility improvements: Button elements & a rel attributes
## requirements
copy pasted from [this PR](#43):

1. Change the default close button & icon elements to use `<button>` elements, either instead of a div, or as an added wrapper around the `<i>` icon element.
This is mainly to make these elements focusable, tabbable and clickable with a keyboard. Among other things, this change will improve accessibility for visually-impaired screenreader users, as well as users with reduced motor skills who might prefer to use a keyboard instead of a mouse.
2. Add 'rel' attribute param to the link option. Allowing developers to add the rel attribute will enable them to set rel="noopener noreferrer" on external links, especially those which open in a new tab. This helps mitigate a [security](https://mathiasbynens.github.io/rel-noopener/) [vulnerability](https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/) and [potentially improves performance](https://jakearchibald.com/2016/performance-benefits-of-rel-noopener/) too.
3. Improve contrast of button foreground/background colours, to help meet [WCAG text contrast guidelines](https://webaim.org/resources/contrastchecker/).

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by richardwestenra**
Closes #45

## Description
1. Change the default close button & icon elements to use `<button>` elements, either instead of a div, or as an added wrapper around the `<i>` icon element.
This is mainly to make these elements focusable, tabbable and clickable with a keyboard. Among other things, this change will improve accessibility for visually-impaired screenreader users, as well as users with reduced motor skills who might prefer to use a keyboard instead of a mouse.
2. Add 'rel' attribute param to the link option. Allowing developers to add the rel attribute will enable them to set rel="noopener noreferrer" on external links, especially those which open in a new tab. This helps mitigate a [security](https://mathiasbynens.github.io/rel-noopener/) [vulnerability](https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/) and [potentially improves performance](https://jakearchibald.com/2016/performance-benefits-of-rel-noopener/) too.
3. Improve contrast of button foreground/background colours, to help meet [WCAG text contrast guidelines](https://webaim.org/resources/contrastchecker/).

## Notes
1. I've tried to keep the styling as close as possible to the existing styling. I've retained the default focus outline (which is very important for keyboard a11y). I would've liked to add extra hover/focus/active styles for the buttons but I'm not sure how to do that with the existing inline style setup, plus that's probably best kept for another PR.
2. I've made the rel attribute a string prop so developers can set whatever rel content they want, in case they want to set something else like rel="external" or rel="home" or whatever.

### CODE REVIEW

**Comment by FrancescoCioria** [comment]
Hi @richardwestenra and thanks for this great PR!

The code looks 👍 
I'll try it locally asap

**Comment by richardwestenra** [comment]
Thanks @FrancescoCioria! Note that I just noticed and fixed a missing comma. Hopefully that'll fix the broken build?

**Comment by FrancescoCioria** [comment]
> Hopefully that'll fix the broken build?

that was definitely a mistake, but the CI is failing for other reasons and I don't think it's because of your PR... I'll look into it and see if I can fix it

File: src/BannerContent.tsx [hunk 1]
```diff
@@ -13,7 +13,8 @@ export type Props = {
   link?: {
     msg?: string,
     url: string,
-    target?: '_blank' | '_self' | '_parent' | '_top' | 'framename'
+    target?: '_blank' | '_self' | '_parent' | '_top' | 'framename',
+    rel: string,
```
Comment (FrancescoCioria) [inline src/BannerContent.tsx]: this should be optional, you can mark it as optional by adding the `?` before the colon

Comment (richardwestenra) [reply src/BannerContent.tsx]: @FrancescoCioria good catch, thanks! I've just fixed it.

**Comment by FrancescoCioria** [comment]
I tested and it looks fine 🎉 💅

**Pull request merged**